### PR TITLE
Keep Login/Signup Form Values on Resize

### PR DESCRIPTION
**Pull request by SofieDunt**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/925293934/pulses/1012460652)

Prevents form fields from being cleared on window resize.

## This PR

- Used `Form.useForm` to keep form instance consistent across forms rendered on different window types
- Passed `windowType` to forms as a prop rather than having both pages and forms use the `useWindowDimensions` hook (less event listeners)

## Screenshots

 
No styling changes.

## Verification Steps

 
Ran locally, input values into form fields and resized the browser past each breakpoint to check that
- layout updated according to designs
- values did not clear past breakpoints

### CODE REVIEW

File: src/components/signupForm/index.tsx [hunk 1]
```diff
@@ -24,18 +25,23 @@ const Footer: typeof Paragraph = styled(Paragraph)<ParagraphProps>`
...
+}) => {
   return (
     <>
-      <Form name="basic" initialValues={{ remember: true }} onFinish={onFinish}>
+      <Form name="basic" form={formInstance} onFinish={onFinish}>
         <FullWidthSpace direction="vertical" size={1}>
           <FormRow>
             <FormHalfItem
+              name="firstName"
               rules={[
```
Comment (willmt80) [inline src/components/signupForm/index.tsx:45]: 💭  Maybe create a form rules file with all of the rules we use throughout the app.

**Review by willmt80 — CHANGES_REQUESTED** [review]
Looks good! left a suggestion about reusing the form item rule logic

File: src/components/signupForm/index.tsx (hunk 1 above)
Comment (SofieDunt) [reply src/components/signupForm/index.tsx:45]: Yeah, I thought about it but the only rules in common between the login/signup form is the one for email. The signup form has more in common with the settings form, so I was just going to pull out the common ones in the ticket to fix settings. You're right though, I'll create a `formRules` file as part of this and update settings later. Do you think we should put all of the rules in there or just the ones used in multiple files?

**Review by jackblanc — APPROVED** [review]
👍

**Review by rymaju — APPROVED** [review]
Looks pretty straightforward 💯

**Comment by willmt80** [comment]
💯

**Review by willmt80 — APPROVED** [review]

**Pull request merged**